core/cloudflow-operator/src/main/scala/cloudflow/operator/CloudflowApplication.scala

```diff
@@ -233,7 +235,15 @@ object CloudflowApplication {
     }
 
     def toAction(app: CloudflowApplication.CR): Action[ObjectResource] =
-      Action.updateStatus(app.withStatus(this), editor)
+      Action.updateStatus(
+        app.withStatus(this),
+        editor,
+        (oldApp: Option[CloudflowApplication.CR], _: CloudflowApplication.CR) =>
+          oldApp.flatMap(_.status) match {
+            case Some(status) if status.appStatus == Some(Status.Error) => false
+            case _                                                      => true
+          }
```

Comment:
This is very dense. Maybe find a good place and name for it?
`ignoreOnErrorStatus`, `requireOkStatus` ?

Reply:
The logic here is already pretty self-contained, maybe I'm missing the reason why this has been implemented as an additional argument and not directly in the `UpdateStatusAction` (e.g. [here](https://github.com/lightbend/cloudflow/pull/830/files#diff-2401f079f994e5a152de63295fa322693b201133df858e2d361ad912b765cd5bR313) )

I think this will automatically make the code less dense as well 😄

Reply:
`UpdateStatusAction` is for any `T <: ObjectResource`, not only for `CloudflowApplication`, which is why it is not implemented there.

Reply:
I'm not totally happy with the results of this so I might keep it a bit ugly :-)

Reply:
I refactored it into `ignoreOnErrorStatus`.